code/go/internal/validator/folder_item_spec.go

```diff
@@ -14,15 +14,16 @@ import (
 )
 
 type folderItemSpec struct {
-	Description      string `yaml:"description"`
-	ItemType         string `yaml:"type"`
-	ContentMediaType string `yaml:"contentMediaType"`
-	Name             string `yaml:"name"`
-	Pattern          string `yaml:"pattern"`
-	Required         bool   `yaml:"required"`
-	Ref              string `yaml:"$ref"`
-	Visibility       string `yaml:"visibility" default:"public"`
-	commonSpec       `yaml:",inline"`
+	Description       string   `yaml:"description"`
+	ItemType          string   `yaml:"type"`
+	ContentMediaType  string   `yaml:"contentMediaType"`
+	ForbiddenPatterns []string `yaml:"forbiddenPatterns"`
```

Comment:
If we can _somehow_ make the regular expression work (per our conversation off-PR), we wouldn't need this field, right?

For now, could you move it to below the `Pattern` field in the struct since they are related.

Reply:
Yes, we wouldn't need it.

Apart from that, it's MUCH simpler and easier to read, having two independent patterns. See: https://github.com/elastic/elastic-package/pull/154#discussion_r515173374